### PR TITLE
fix(contracts): `RateLimit` minor changes

### PR DESCRIPTION
**Pull request by aroralanuk**
### Description

- Added a check for invalid capacity and event for token level change

### Drive-by changes

None

### Related issues

- fixes https://github.com/chainlight-io/2024-08-hyperlane/issues/14

### Backward compatibility

Yes

### Testing

Unit tests

### CODE REVIEW

File: solidity/contracts/libs/RateLimited.sol [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
 contract RateLimited is OwnableUpgradeable {
     uint256 public constant DURATION = 1 days; // 86400
-    uint256 public filledLevel; /// @notice Current filled level
-    uint256 public refillRate; /// @notice Tokens per second refill rate
-    uint256 public lastUpdated; /// @notice Timestamp of the last time an action has been taken TODO prob can be uint40
+    /// @notice Current filled level
+    uint256 public filledLevel;
+    /// @notice Tokens per second refill rate
+    uint256 public refillRate;
+    /// @notice Timestamp of the last time an action has been taken TODO prob can be uint40
```
Comment (yorhodes) [inline solidity/contracts/libs/RateLimited.sol:32]:
```suggestion
    /// @notice Timestamp of the last time an action has been taken
```
remove TODO?

File: solidity/contracts/libs/RateLimited.sol [hunk 2]
```diff
@@ -88,20 +112,22 @@ contract RateLimited is OwnableUpgradeable {
...
-        require(_newAmount <= adjustedFilledLevel, "RateLimitExceeded");
+        require(_consumedAmount <= adjustedFilledLevel, "RateLimitExceeded");
 
         // Reduce the filledLevel and update lastUpdated
-        uint256 _filledLevel = adjustedFilledLevel - _newAmount;
+        uint256 _filledLevel = adjustedFilledLevel - _consumedAmount;
         filledLevel = _filledLevel;
         lastUpdated = block.timestamp;
 
+        emit ConsumedFilledLevel(filledLevel, lastUpdated);
```
Comment (yorhodes) [inline solidity/contracts/libs/RateLimited.sol:129]: whats the intended use for this?

**Review by yorhodes — APPROVED** [review]
audit remediation is a funny time to introduce ascii art lol

File: solidity/contracts/libs/RateLimited.sol (hunk 2 above)
Comment (aroralanuk) [reply solidity/contracts/libs/RateLimited.sol:129]: Part of the audit, to have events for state changes.

Comment (yorhodes) [reply solidity/contracts/libs/RateLimited.sol:129]: personally dont think this applies pervasively but okay

**Pull request merged**